bagvalidate/activity.go

```diff
@@ -44,14 +43,14 @@ func (a *Activity) Execute(ctx context.Context, params *Params) (*Result, error)
 	logger.V(1).Info("Executing bag-validate activity", "Path", params.Path)
 
 	if err := a.validator.Validate(params.Path); err != nil {
-		if errors.Is(convertError(err), ErrInvalid) {
-			return &Result{
-				Valid: false,
-				Error: err.Error(),
-			}, nil
+		wrappedErr := errors.Unwrap(err)
+
+		// Handle application errors.
+		if wrappedErr != ErrNotABag && wrappedErr != ErrInvalid {
```

Comment:
I think the idiomatic Go way to handle checking for a wrapped error type is with `errors.Is()` which checks the whole error tree (unwrapping each wrapped error). I believe:

```go
if !errors.Is(err, ErrNotABag) && !errors.Is(err, ErrInvalid) {
```

will function the same as the above conditional without the need to define `wrappedErr`.